controllers/cassandrarestore_controller.go

```diff
@@ -172,6 +177,39 @@ func (r *CassandraRestoreReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 			return ctrl.Result{RequeueAfter: 10 * time.Second}, err
 		}
 
+		afterHash := deepHashString(cassdc.Spec.PodTemplateSpec)
+
+		if restore.Spec.Shutdown {
+			// cass-operator does not apply updates to the StatefulSet's template spec until
+			// pods are ready. This means that when we shutdown and update the template spec
+			// with the backup name and the restore key, cass-operator first scales the
+			// StatefulSet back up and then only after the pods are ready does it update the
+			// template spec. This results in a StatefulSet update which has the effect of a
+			// cluster rolling restart. The following if block is a work around to avoid that
+			// rolling restart.
+			if beforeHash != afterHash {
+				racks := make([]string, 0, len(cassdc.Spec.Racks))
+				for _, rack := range cassdc.Spec.Racks {
+					racks = append(racks, rack.Name)
+				}
+				cassdc.Spec.ForceUpgradeRacks = racks
+
+				r.Log.Info("updating racks", "CassandraDatacenter", cassdcKey, "Racks", cassdc.Spec.ForceUpgradeRacks)
+
+				if err = r.Update(ctx, cassdc); err == nil {
+					return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+				}
+
+				r.Log.Error(err, "failed to force update racks", "CassandraDatacenter", cassdcKey)
+				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+			} else {
```

Comment:
You'd really want gostaticcheck or equal (VSCode autoinstalls it) to avoid these unnecessary else clauses ;)

Reply:
I took your suggestion and ran `staticcodecheck`. It did not report anything on that else clause. I'll go ahead and remove it. While unnecessary, I sometimes find that with larger conditional blocks the explicit else improves readability.